### ISSUE TITLE
Is there an API ?

### ISSUE THREAD

**Issue by JamesDAdams** [retspen/webvirtcloud#466]
Hi,

i use Webvirtcloud for start my vm, but i would like start a script when the server power on request is sent and when it is off would it be possible to have an api for that ?

Tanks.

**Comment by catborise** [comment]
do you make something on vm’s OS with script?

**Comment by JamesDAdams** [comment]
No, I can't, because I want to run a script before the vm starts, if that is not possible where is the webvirtcloud log file located ? Because I can do a bash script that checks the log file every x seconds to see if a vm has been started

**Comment by JamesDAdams** [comment]
Hi,

Is there a log file that displays the same logs as on the logs web page ?

Thanks.

**Comment by catborise** [comment]
the logs does not write to the log files only to database but we can write them also to files.

**Comment by catborise** [comment]
With Django REST Framework, REST API is implemented. for now; 
compute, instance : list, create, update, delete operations can be performed.
also instance, power on/off, etc. operations can be done with rest api.
you can check http://<webvirtcloud_installaiton_address:port>/swagger, http://<webvirtcloud_installaiton_address:port>/redoc to see capabilities.
